### PR TITLE
URL request constructor extension

### PR DESCRIPTION
**Pull request by andrewjl-mux**
Pulled URL request constructor change into a separate PR. Besides moving logic into an extension, here the SDK avoids sending a request without a serialized body.

This latter fix is consistent with #49. In addition to this fix the other PR has a deeper cleanup across the entire processing pipeline.

### CODE REVIEW

File: Sources/MuxUploadSDK/Extensions/File.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//
+//  NSMutableURLRequest+Reporting.swift
+//  
+
+import Foundation
+
+extension NSMutableURLRequest {
+    static func make(
```
Comment (daytime-em) [inline Sources/MuxUploadSDK/Extensions/File.swift:8]: nit, nonblocking: 
```suggestion
    static func makeJSONPost(
```

**Review by daytime-em — APPROVED** [review]
LGTM, just had a nit

**Comment by andrewjl-mux** [comment]
> LGTM, just had a nit

Great suggestion for a more specific name, probably fixed a future bug

**Comment by andrewjl-mux** [comment]
Somehow the file added was still `File.swift` so renamed it to match what's inside of it.

**Pull request merged**